doc/scaling_spire.md

```diff
@@ -0,0 +1,43 @@
+
+# Topologies
+
+There are three main SPIRE deployment topologies; single trust domain, nested SPIRE, and federated SPIRE. Multiple considerations should are to be factored in when choosing one topology design over another.
+
+A common motivation for a single overarching trust domain is to issue identifies from a single Certificate Authority, as it reduces the number of components to manage. However, when deploying a single SPIRE trust domain to span regions and clouds. 
+Managing a shared datastore across geographically dispersed locations or across cloud providers may represent a challenge. In such circumstances, a solution is to configure SPIRE servers can be configured in Nested topology.
+
+## Nested SPIRE
+
+Nested SPIRE allows SPIRE servers to be chained together, and for all servers to still issue identities in the same trust domain. It works by co-locating an agent with downstream servers. Each downstream server loads the CA credentials from the Workload API of the top or global SPIFFE implementation. The downstream node performs both node and workload attestation, which in turn allows automating the bring-up of new downstream servers. 
+ 
+This topology is best suited for multi-cloud deployments. It can be reasoned as the top server being a global server (or set of servers for high availability). The downstream servers may be deployed at either regional or cluster level. Due to the ability to mix and match node attestors, the downstream servers can reside in different clouds. 
+ 
+In this configuration, the top tier SPIRE servers hold the root certificate/key, and the downstream servers request an intermediate signing certificate to use as the server's X.509 signing authority. It provides for resilience as the top tier can go down, and intermediate certificates will continue to operate. 
+ 
+Complimentary to scaling SPIRE servers horizontally to deal with load, a nested topology helps to segment failure domains.
+
+## Federated SPIRE
+
+A requirement or objective may exist to make use of multiple roots of trust: perhaps because an organization has different organizational divisions with different administrators, or because they have separate staging and production environments that occasionally need to communicate. A similar use-case is SPIFFE interoperability between organizations, such as between a cloud provider and its customers. Both use-cases require a well-defined, interoperable method for a workload in one trust domain to authenticate a workload in a different trust domain.
+Trust between the different trust domains is established by authenticating the respective Bundle Endpoint. For additional detail on how this is achieved, refer to the following SPIFFE spec that describe the mechanism: 
```

Comment:
Authenticating the endpoint is the first step. Obtaining the foreign trust domain bundle via the authenticated endpoint is the next.

Reply:
Updated to outline the order of operations.